### ISSUE TITLE
Should decoder discard messages with invalid address?

### ISSUE THREAD

**Issue by DrLuke** [klingtnet/rosc#25]
I've been trying to implement a dispatcher and noticed that the current matcher is a bit unergonomic to use. The `OscMessage`'s address is only checked for validity in the matcher. This results in having to check for address errors during address matching, which seems like the wrong place to do that, as a dispatcher will attempt to match with every OSC method, but every single one will fail due to the address being invalid, so the same work is done repeatedly.

I propose that we instead discard incoming `OscMessage`s in the decoder, as an `OscMessage` without a valid address can not ever be matched anyway, and as a consequence always know that an `OscMessage` must have a valid address.

I wanted to gather thoughts on this before implementing it.

**Comment by klingtnet** [comment]
> I propose that we instead discard incoming OscMessages in the decoder, as an OscMessage without a valid address can not ever be matched anyway, and as a consequence always know that an OscMessage must have a valid address.

I like the idea but the address check should be optional, i.e. configurable.  OSC does specify an address pattern syntax but since they're only strings one can match however they like.  So, I think it would be sensible to discard messages with invalid addresses by default but still allow to explicitly disable this check.  What do you think?

**Comment by DrLuke** [comment]
Very good point, I didn't think about that! I'll whip up a PR after finishing my current PR.

**Comment by klingtnet** [comment]
Thanks for raising the question and I'm looking forward to the PR!

**Comment by eugenwintersberger** [comment]
Are there any plans to implement this? It would be great to turn off address checks. I am actually trying to write some code to control Ardour via OSC> It seems that Ardour uses `#reply`  as an address in some responses which currently does not work with `rosc`.

**Comment by klingtnet** [comment]
No, there's no work planned, but I'm to review and merge if someone wants to implement conditional address checks.
